### PR TITLE
Jjhursey topic/timer gettimeofday

### PR DESCRIPTION
**Pull request by hppritcha**
add some PR feedback to @jjhursey 's original PR #3184 

@jsquyres this is it.  have commitment this evening so its this or wait till tomorrow.

### CODE REVIEW

File: ompi/mpi/c/wtick.c [hunk 1]
```diff
@@ -40,6 +41,12 @@ double MPI_Wtick(void)
 {
     OPAL_CR_NOOP_PROGRESS();
 
+    /*
+     * See https://github.com/open-mpi/ompi/issues/3003
+     * For now we are forcing the use of gettimeofday() until we find a
```
Comment (jsquyres) [inline ompi/mpi/c/wtick.c]: This comment is no longer correct.

File: ompi/mpi/c/wtick.c [hunk 2]
```diff
@@ -52,8 +59,14 @@ double MPI_Wtick(void)
     }
 #elif OPAL_TIMER_USEC_NATIVE
     return 0.000001;
+#endif
+#else
+#if defined(__linux__) && OPAL_HAVE_CLOCK_GETTIME
+    /* QAD no call clock_getres */
```
Comment (jsquyres) [inline ompi/mpi/c/wtick.c]: What's "QAD" -- why not call `clock_getres`?

File: ompi/mpi/c/wtime.c [hunk 3]
```diff
@@ -40,17 +44,30 @@ double MPI_Wtime(void)
...
 #if OPAL_TIMER_CYCLE_NATIVE
     wtime = ((double) opal_timer_base_get_cycles()) / opal_timer_base_get_freq();
 #elif OPAL_TIMER_USEC_NATIVE
     wtime = ((double) opal_timer_base_get_usec()) / 1000000.0;
+#endif
+#else
+#if defined(__linux__) && OPAL_HAVE_CLOCK_GETTIME
+    struct timespec tp = {.tv_sec = 0, .tv_nsec = 0};
+    (void) clock_gettime(CLOCK_MONOTONIC, &tp);
+    return (tp.tv_sec + tp.tv_nsec/1.0e+9);
```
Comment (jsquyres) [inline ompi/mpi/c/wtime.c]: This `return` is not quite right -- all paths through the original code assign `wtime` and fall through to call `OPAL_CR_NOOP_PROGRESS`.

I know the checkpoint/restart code has fallen into disrepair, but the disparity of treatment here is a little odd.

File: ompi/mpi/c/wtick.c (hunk 2 above)
Comment (hppritcha) [inline ompi/mpi/c/wtick.c]: quick and dirty

**Comment by hppritcha** [comment]
@jsquyres updated per your feedback.  the original merge commit was showing up because I did a merge of @jjhursey stuff on to a fork of master - i.e. doing github PR like thing by hand.

**Pull request merged**